js/style/parse_color.js

```diff
@@ -1,46 +1,40 @@
 'use strict';
 
-var parseCSSColor = require('csscolorparser').parseCSSColor;
+var parseColorString = require('csscolorparser').parseCSSColor;
 var util = require('../util/util');
+var StyleFunction = require('./style_function');
 
-var colorCache = {};
+var cache = {};
 
-function parseColor(input) {
+module.exports = function parseColor(input) {
 
-    if (colorCache[input]) {
-        return colorCache[input];
+    if (StyleFunction.isFunctionDefinition(input)) {
 
-    // RGBA array
-    } else if (Array.isArray(input)) {
-        return input;
-
-    // GL function
-    } else if (input && input.stops) {
         return util.extend({}, input, {
-            stops: input.stops.map(parseFunctionStopColor)
+            stops: input.stops.map(function(stop) {
+                return [stop[0], parseColor(stop[1])];
+            })
```

Comment:
Inlining this function improves readability. This code is not hot enough to warrant sacrificing readability for performance. For example, this code path only runs 33 times on our debug page which renders mapbox streets.